src/main/java/com/example/b2b_opportunities/Service/PasswordService.java

```diff
@@ -0,0 +1,45 @@
+package com.example.b2b_opportunities.Service;
+
+import com.example.b2b_opportunities.Config.SecurityConfig;
+import com.example.b2b_opportunities.Dto.Request.ResetPasswordDto;
+import com.example.b2b_opportunities.Entity.ConfirmationToken;
+import com.example.b2b_opportunities.Entity.User;
+import com.example.b2b_opportunities.Exception.DisabledUserException;
+import com.example.b2b_opportunities.Exception.PasswordsNotMatchingException;
+import com.example.b2b_opportunities.Repository.ConfirmationTokenRepository;
+import com.example.b2b_opportunities.Repository.UserRepository;
+import jakarta.servlet.http.HttpServletRequest;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class PasswordService {
+    private final MailService mailService;
+    private final UserRepository userRepository;
+    private final AuthenticationService authenticationService;
+    private final ConfirmationTokenRepository confirmationTokenRepository;
+
+    public String requestPasswordRecovery(String email, HttpServletRequest request) {
+        User user = userRepository.findByEmail(email).orElseThrow(() -> new NullPointerException("User not registered"));
+        if (!user.isEnabled()) {
+            throw new DisabledUserException("Account not activated");
+        }
```

Comment:
Add an additional check if the user is registered with oAuth. There's no need to send him a password.